docs/superpower/CCCT-2418-tech-spec-flatten.md

```
@@ -0,0 +1,167 @@
+**Migration 1 — Schema:** Add `program` to `Opportunity`.
+
+* Rename the `ManagedOpportunity.program` field to `program_old` and make it nullable.
+* Add a new `program` field directly to `Opportunity`.
+* Backfill the new `Opportunity.program` field from the existing `ManagedOpportunity.program_old` values.
+
+  This approach minimizes code changes. The primary impact is that reverse relationship queries will need to be updated from `program__managedopportunity` to `program__opportunity`. We currently use this reverse relationship in only a few places (4), so the required code changes should be small. All other access patterns should continue to work with little or no modification.
+
+    > [!NOTE]
+    > The `claimed` field can be ignored during the migration, as it is not currently used anywhere in the codebase. It can be dropped as part of the cleanup.
+
+----
```

markdownlint-cli2 (0.22.1)

[warning] Horizontal rule style
Expected: ---; Actual: ----

(MD035, hr-style)